src/udev/udev-builtin-net_id.c

```diff
@@ -414,21 +428,26 @@ static int names_ccw(struct  udev_device *dev, struct netnames *names) {
         struct udev_device *cdev;
         const char *bus_id;
         size_t bus_id_len;
+        size_t bus_id_start;
         int rc;
+        char *subsys;
 
         assert(dev);
         assert(names);
 
         /* Retrieve the associated CCW device */
         cdev = udev_device_get_parent(dev);
+        /* skip virtio subsystem if present */
+        cdev = skip_virtio(cdev);
         if (!cdev)
                 return -ENOENT;
 
-        /* Network devices are always grouped CCW devices */
-        if (!streq_ptr("ccwgroup", udev_device_get_subsystem(cdev)))
+        /* Network devices are either single or grouped CCW devices */
+        subsys = udev_device_get_subsystem(cdev);
+        if (!STR_IN_SET(subsys, "ccwgroup", "ccw"))
```

Comment:
NP, but while we're at it: as the STRPTR_IN_SET evaluates the first argument just once, would you prefer that I also drop the subsys variable, that I mainly introduced to avoid multiple u_d_g_s() calls?